### PR TITLE
Add `is-prerelease script`, refactor

### LINKED ISSUE #19: Program to know if git tag should be treated as a prerelease
If git tag starts with `v0`, e.g `v0.0.5`, it is a "prerelease" version and GitHub release created from it should be marked as such.

We also want to handle prefixes, such as `comms-v0.0.5`.

Currently it's done in an ugly way in GitHub Actions directly:

```bash
- name: Set version data
  run: |
      tag=${{ github.ref_name }}
      prefix=comms-v
      echo "RELEASE_NOTES=$(link-changelog comms $tag $prefix)" >> $GITHUB_ENV
      # set whether is prerelease
      version=${tag#"$prefix"}
      is_prerelease=false
      if [[ "$version" == 0.* ]]; then
        is_prerelease=true
      fi
      echo "IS_PRERELEASE=$is_prerelease" >> $GITHUB_ENV

- name: Create release
   uses: softprops/action-gh-release@v1
   with:
     name: ${{ github.ref_name }}
     body: ${{ env.RELEASE_NOTES }}
     prerelease: ${{ env.IS_PRERELEASE }}
```

Is there an easy way to do this or should I write a small `is-prerelease` shellscript, similarly to `link-changelog`? cc: @iasiu, @Saancreed

### LINKED ISSUE #20: Remove the need to specify git tag prefix manually
[Originated from here](https://github.com/leancodepl/mobile-tools/issues/19#issuecomment-1230623443).

### PR DESCRIPTION
**Pull request by bartekpacia**
Fixes #19 and fixes #20.

### CODE REVIEW

File: bin/is-prerelease [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+fi
+
+if [ -z "$tag" ]; then
+    echo "error: missing tag" 1>&2
+    exit 1
+fi
+
+prefix="${package_name}-"
+version="${tag#$prefix}"
+version="${version#v}"
```
Comment (bartekpacia) [inline bin/is-prerelease:35]: is such "staged" processing considered OK and idiomatic, @Saancreed?

File: bin/is-prerelease [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+if [[ "$version" == 0* ]]; then
+    echo "true"
+else
+    echo "false"
+fi
```
Comment (bartekpacia) [inline bin/is-prerelease:41]: Is using `true` and `false` strings okay?

File: bin/is-prerelease (hunk 1 above)
Comment (Saancreed) [reply bin/is-prerelease:35]: Yeah, it makes sense to do it like this.

File: bin/is-prerelease (hunk 2 above)
Comment (Saancreed) [reply bin/is-prerelease:41]: Why not? There is no convention here afaict so the way the tools like this one return their results is all over the place.

* `test` and various similar builtins like `[` and `[[` return `0` as exit code on success and `1` on failure
* [`vercmp`](https://man.archlinux.org/man/vercmp.8) prints `-1` / `0`/ `1` to stdout
* `systemctl is-…` both prints a string and returns a meaningful exit code unless `--quiet` is specified:

```
$ systemctl is-system-running 
degraded

$ echo $?
1
```

My point is, whatever you do, it'll probably be fine.

File: bin/link-changelog [hunk 3]
```diff
@@ -19,10 +30,9 @@ if [ -z "$tag" ]; then
     exit 1
 fi
 
-if [ -n "$prefix" ]; then
-    tag="${tag#$prefix}"
-fi
-
-version=$(echo "${tag#v}" | tr -d '.+')
+prefix="${package_name}-"
```
Comment (Saancreed) [inline bin/link-changelog:33]: Tbh you could probably keep taking it from `$3` if explicitly specified:
```suggestion
prefix="${3:-${package_name}-}"
```

**Review by Saancreed — APPROVED** [review]

**Pull request merged**